### PR TITLE
StaticLLMPipeline: Use set_tensor for kvcache model

### PR DESCRIPTION
**Pull request by TolyaTalamanov**

### CODE REVIEW

File: src/cpp/src/llm_pipeline_static.cpp [hunk 1]
```diff
@@ -704,6 +704,17 @@ EncodedResults StaticLLMPipeline::generate(
         position_ids_data[0] = m_kvcache_desc.num_stored_tokens;
         attention_mask_data[m_kvcache_desc.num_stored_tokens - 1] = 1u;
 
+        // NB: Write KV-cache for the new token to the correct input position for the next iteration
```
Comment (TolyaTalamanov) [inline src/cpp/src/llm_pipeline_static.cpp:707]: Comment should be changed

**Comment by dmatveev** [comment]
@TolyaTalamanov did it change anything?

File: src/cpp/src/llm_pipeline_static.cpp (hunk 1 above)
Comment (soumendukrg) [reply src/cpp/src/llm_pipeline_static.cpp:707]: @TolyaTalamanov Does set_tensor now support remote tensor data sharing across CPU and NPU?

**Comment by TolyaTalamanov** [comment]
> @TolyaTalamanov did it change anything?

It broke accuracy, although on CPU it works fine. Will re-consider this optimization

**Comment by ilya-lavrenov** [comment]
Please, rebase to latest master
